plaidml/bridge/openvino/ops/detection_output.cpp

```diff
@@ -0,0 +1,464 @@
+    }
+
+    // Transpose the confidence to match the input shape of `scores` in NMS.
+    // confidence -> {batch, num_boxes, num_classes}
+    // scores -> {batch, num_classes, num_boxes}
+    edsl::Tensor nms_conf = op::transpose(confidence, edsl::make_tuple<int64_t>({0, 2, 1}));
+    // Remove the background class.
+    if (background_label_id > -1) {
+      std::vector<edsl::Tensor> idx_vec;
+      edsl::Tensor idx_first_half = edsl::index({edsl::TensorDim(background_label_id)}, 0);
+      edsl::Tensor idx_second_half =
+          op::slice(edsl::index({edsl::TensorDim(num_classes)}, 0)).add_dim(background_label_id + 1, num_classes);
+      idx_vec.push_back(idx_first_half);
+      idx_vec.push_back(idx_second_half);
+      edsl::Tensor gather_idxs;
+      if (background_label_id != 0) {
+        gather_idxs = op::concatenate(idx_vec, 0);
+      } else {
+        gather_idxs = idx_second_half;
+      }
+      nms_conf = edsl::gather(nms_conf, gather_idxs).axis(1);
+    }
+
+    // result = {selected_indices, selected_scores, valid_outputs}
+    edsl::Tensor iou_threshold = cast(edsl::Tensor{nms_threshold}, DType::FLOAT32);
+    edsl::Tensor score_threshold = cast(edsl::Tensor{confidence_threshold}, DType::FLOAT32);
+    std::vector<edsl::Tensor> result =
+        nms(decoded_bboxes, nms_conf, top_k, iou_threshold, score_threshold, cast(edsl::Tensor{0}, DType::FLOAT32))
+            .center_point_box(false)
+            .sort_result_descending(false)
+            .box_input_type(DType::FLOAT32)
+            .thres_type(DType::FLOAT32)
+            .build();
+    edsl::Tensor selected_indices = result[0];
+    auto selected_indices_shape = selected_indices.compute_shape().sizes();
+    edsl::Tensor selected_scores = result[1];
+    edsl::Tensor valid_outputs = result[2];
+
+    // TODO
+    edsl::Tensor batch_slice = op::slice(selected_indices).add_dim(0, selected_indices_shape[0]).add_dim(0, 1);
+    edsl::Tensor indices_slice = op::slice(selected_indices).add_dim(0, selected_indices_shape[0]).add_dim(2, 3);
+    std::vector<edsl::Tensor> gather_idxs_vec = {batch_slice, indices_slice};
+    edsl::Tensor gather_idxs = cast(op::concatenate(gather_idxs_vec, 1), DType::INT32);
+    edsl::Tensor out_boxes = edsl::gather(decoded_bboxes, gather_idxs).mode(GatherMode::ND);
+
+    if (clip_after_nms) {
+      out_boxes = op::maximum(cast(edsl::Tensor{0.0f}, DType::FLOAT32),
+                              op::minimum(cast(edsl::Tensor{1.0f}, DType::FLOAT32), out_boxes));
+    }
+
+    edsl::Tensor class_slice = op::slice(selected_indices).add_dim(0, selected_indices_shape[0]).add_dim(1, 2);
+    auto one = cast(edsl::Tensor{1}, class_slice.dtype());
+    auto slice_shape = class_slice.compute_shape().sizes();
+    std::vector<int> target_shape(begin(slice_shape), end(slice_shape));
+    std::vector<int> target_axes = {};
+    auto one_slice = op::broadcast(one, target_shape, target_axes);
```

Comment:
Same as above, no need to cast to `vector<int>`. Please check.

Reply:
Done.